src/hotspot/share/opto/memnode.cpp

```diff
@@ -3233,6 +3267,15 @@ Node* MergePrimitiveStores::make_merged_input_value(const Node_List& merge_list)
          (_phase->type(merged_input_value)->isa_long() != nullptr && new_memory_size == 8),
          "merged_input_value is either int or long, and new_memory_size is small enough");
 
+  if (_value_order == DataOrder::Reverse) {
```

Comment:
```suggestion
  if (_value_order == DataOrder::Reverse) {
    assert(_store->memory_size() == 1, "only implemented for bytes");
```
That would be correct, right?